src/agentscope/service/web_search/web_digest.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+"""parsing and digesting the web pages"""
+from typing import Optional
+from langchain_community.document_loaders import AsyncChromiumLoader
+from langchain_community.document_transformers import BeautifulSoupTransformer
+from langchain_text_splitters import HTMLHeaderTextSplitter
+
+from agentscope.service.service_response import ServiceResponse
+from agentscope.service.service_status import ServiceExecStatus
+from agentscope.models.model import ModelWrapperBase
+from agentscope.message import Msg
+
+
+DEFAULT_SYS_PROMPT = (
+    "You're a data cleaner. You job is to extract useful "
+    "training data from html or webpage source.\n"
+    "Extract useful part from the following data:{}"
+)
+
+DEFAULT_SPLIT_LEVEL = [
+    ("h1", "Header 1"),
+    ("h2", "Header 2"),
+]
+
+
+def webpage_digest(
+    url: str,
+    model: Optional[ModelWrapperBase] = None,
+    html_split_levels: Optional[list[tuple[str, str]]] = None,
+    digest_prompt: Optional[str] = DEFAULT_SYS_PROMPT,
+) -> ServiceResponse:
+    """
+    Parses the web page
+    :param url: the url of the web page
+    :param model: the model that is used to digest the web page content.
+    :param html_split_levels:  required parameter for HTMLHeaderTextSplitter
+    :param digest_prompt: prompt for the model to analyze the webpage content
+
+    :return: a ServiceResponse containing the digested content
+
+    If the model is None, then the webpage content will be processed
+    by **BeautifulSoupTransformer**.
```

Comment:
In the docstring, it's unclear how the model and beautifulsoup4 will handle the web pages, e.g. what's the output of this function?

Reply:
updated